nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySObject.java

```diff
@@ -0,0 +1,499 @@
+                JsonTreeRowRecordReader jsonReader = new JsonTreeRowRecordReader(
+                    querySObjectResultInputStream,
+                    getLogger(),
+                    convertedSalesforceSchema.querySObjectResultSchema,
+                    dateFormat,
+                    timeFormat,
+                    timestampFormat
+                );
+
+                RecordSetWriter writer = writerFactory.createWriter(
+                    getLogger(),
+                    writerFactory.getSchema(
+                        originalAttributes,
+                        convertedSalesforceSchema.recordSchema
+                    ),
+                    out,
+                    originalAttributes
+                );
+            ) {
+                writer.beginRecordSet();
+
+                Record querySObjectRecord;
+                while ((querySObjectRecord = jsonReader.nextRecord()) != null) {
+                    List<Object> recordObjectList = Arrays.asList(querySObjectRecord.getAsArray("records"));
+                    for (Object recordObject : recordObjectList) {
+                        writer.write((Record) recordObject);
+                    }
+                }
+
+                WriteResult writeResult = writer.finishRecordSet();
+
+                attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                attributes.putAll(writeResult.getAttributes());
+
+                recordCountHolder.set(writeResult.getRecordCount());
+
+                if (ageFilterUpper != null) {
+                    Map<String, String> newState = new HashMap<>(state.toMap());
+                    newState.put(LAST_AGE_FILTER, ageFilterUpper);
+                    try {
+                        context.getStateManager().setState(newState, Scope.CLUSTER);
+                    } catch (IOException e) {
+                        getLogger().error("Couldn't update state", e);
+                        throw new ProcessException(e);
+                    }
+                }
+            } catch (SchemaNotFoundException e) {
+                getLogger().error("Couldn't create record writer", e);
+                throw new ProcessException(e);
+            } catch (MalformedRecordException e) {
+                getLogger().error("Couldn't read records from input", e);
+                throw new ProcessException(e);
+            }
+        });
+
+        int recordCount = recordCountHolder.get();
+
+        if (!includeZeroRecordFlowFiles && recordCount == 0) {
+            session.remove(flowFile);
+        } else {
+            flowFile = session.putAllAttributes(flowFile, attributes);
+            session.transfer(flowFile, REL_SUCCESS);
+
+            session.adjustCounter("Records Processed", recordCount, false);
+            getLogger().info("Successfully written {} records for {}", new Object[]{recordCount, flowFile});
```

Comment:
The `Object[]` wrapper is unnecessary:
```suggestion
            getLogger().info("Processed [{}] records from {}", recordCount, flowFile);
```